config/knexfile.ts

```diff
@@ -1,58 +1,197 @@
+    },
+  },
+  pool: {
+    min: {
+      doc: 'Minimum number of connections',
+      format: 'Number',
+      default: 10
+    },
+    max: {
+      doc: 'Maximum number of connections',
+      format: 'Number',
+      default: 10
+    },
+    acquireTimeoutMillis: {
+      doc: 'How long knex should wait when acquiring a connection before throwing a timeout error',
+      format: 'Number',
+      default: 30000
+    },
+    createTimeoutMillis: {
+      doc: 'How long knex should wait when establishing a new connection before throwing a timeout error',
+      format: 'Number',
+      default: 30000
+    },
+    destroyTimeoutMillis: {
+      doc: 'How long knex should wait when destroying new connection before throwing a timeout error',
+      format: 'Number',
+      default: 5000
+    },
+    idleTimeoutMillis: {
+      doc: 'How long a connection must sit idle in the pool and not be checked out before it is automatically closed',
+      format: 'Number',
+      default: 30000
+    },
+    reapIntervalMillis: {
+      doc: 'How often to check for idle resources to destroy',
+      format: 'Number',
+      default: 1000
+    },
+    createRetryIntervalMillis: {
+      doc: 'How long knex should wait before trying to create a connection again',
+      format: 'Number',
+      default: 200
     }
   },
-  test: {
-    client: 'sqlite3',
-    connection: ':memory:',
-    useNullAsDefault: true,
-    migrations: {
-      directory: migrationsDirectory,
-      tableName: 'auth-service',
-      loadExtensions: ['.ts']
-    },
-    seeds: {
-      directory: seedsDirectory,
-      loadExtensions: ['.ts']
+  migrations: {
+    directory: {
+      doc: 'Migration directory',
+      format: String,
+      default: migrationsDirectory
+    },
+    tableName: {
+      doc: 'Migration table name',
+      format: String,
+      default: 'auth-service'
+    },
+    stub: {
+      doc: 'Where the stubs for migration are located',
+      format: String,
+      default: `${migrationsDirectory}/migration.template`
+    },
+    loadExtensions: {
+      doc: 'Array of extensions to load',
+      format: 'Array',
+      default: ['.ts']
     }
   },
-  production: {
-    /* For now, the production environment is the same as the development environment. */
-    client: 'mysql',
-    version: '5.5',
-    connection: config.get('DATABASE').connection,
-    pool: config.get('DATABASE').pool,
-    migrations: {
-      directory: migrationsDirectory,
-      tableName: 'auth-service',
-      stub: `${migrationsDirectory}/migration.template`,
-      loadExtensions: ['.ts']
-    },
-    seeds: {
-      directory: seedsDirectory,
-      loadExtensions: ['.ts']
+  seeds: {
+    directory: {
+      doc: 'Seeds directory',
+      format: String,
+      default: seedsDirectory
+    },
+    loadExtensions: {
+      doc: 'Array of extensions to load',
+      format: 'Array',
+      default: ['.ts']
     }
   }
-}
+})
+
+const env = process.env.NODE_ENV ?? 'production'
```

Comment:
@eoln I was unable to import the env from the shared Config file. Should I be doing this or should I add the environment variable to the schema here as well?

Reply:
@kkzeng please add the ENV variable to the schema like here: https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/shared/config.ts#L121-L126

Reply:
in definition of format for ENV you should specify the list of allowed values for your environments

Reply:
Right, I wasn't sure if I should duplicate the variable again since in the `~/src/shared/config.ts` we already have the env. So we can add it again to DatabaseConfig schema?

Reply:
no need for ENV in DatabaseConfig -> you are resolving which *_db.ts file to load using the ENV prop defined in ~/src/shared/config

Reply:
So that's what I was saying before - when I import env from ~/src/shared/config, it is `undefined` for some reason. I tried exporting the env variable itself.

So in ~/src/shared/config, I had something like:
`export const env = ConvictConfig.get('ENV')`

Reply:
@kkzeng take a look here: https://github.com/mojaloop/auth-service/blob/master/tsconfig.json#L44
the `~/` is shorthand so you should import from `~/shared/config` and skip `src` part